### PR TITLE
Feature/28 add autologin

### PR DESCRIPTION
**Pull request by easybird**
Added Autologin via AsyncStorage and added Unstated for state management

This PR closes ticket #28

### CODE REVIEW

**Comment by devopsITP** [comment]
:shipit: This branch has been deployed to:
  https://expo.io/@itpocket/white-label-event-mobile-app-feature-28-add-autologin

  Download the [Expo](https://expo.io/) app and scan this QR code to get started!

  QR Code

File: App.js [hunk 1]
```diff
@@ -1,73 +1,43 @@
...
-});
-
 export default class App extends Component<*, State> {
   constructor(props) {
     super(props);
 
     this.firebaseRefs = {};
     this.state = {
-      userInfo: {},
+      shiftData: [],
```
Comment (thibmaek) [inline App.js]: ?

File: App.js [hunk 2]
```diff
@@ -80,49 +50,23 @@ export default class App extends Component<*, State> {
...
-    if (!userInfo.id) return;
-
-    this.setState({
-      userInfo: {
-        ...userInfo,
-        first_name: userInfo.given_name,
-      },
-    });
-  };
+  RootStack: typeof StackNavigator;
```
Comment (thibmaek) [inline App.js]: react-navigation has type declarations you can use instead: https://github.com/react-navigation/react-navigation/blob/master/flow/react-navigation.js

File: App.js [hunk 3]
```diff
@@ -1,73 +1,43 @@
...
-      userInfo: {},
+      shiftData: [],
       usersPerSchedule: {},
     };
 
     StatusBar.setBarStyle('light-content', true);
   }
 
-  componentWillMount() {
+  async componentWillMount() {
```
Comment (thibmaek) [inline App.js]: Not sure why you would mark a method async, if you're not specifically awaiting async ops. Used to work before without the async directive. Why use it now?

File: App.js (hunk 1 above)
Comment (thibmaek) [reply App.js]: Seems like this might be a wrong revert?

File: App.js [hunk 4]
```diff
@@ -80,49 +50,23 @@ export default class App extends Component<*, State> {
...
-    });
-  };
+  RootStack: typeof StackNavigator;
 
   render() {
-    const { userInfo } = this.state;
     return (
-      <View style={styles.container}>
-        <RootStack
+      <Provider inject={[user]}>
```
Comment (thibmaek) [inline App.js]: 🙌

File: App.js (hunk 1 above)
Comment (thibmaek) [inline App.js]: Seems like this is something wrong that came along a revert? Anyway, local state type should be updated as well if this is correct.

File: screens/login/LoginContainer.js [hunk 5]
```diff
@@ -1,32 +1,37 @@
...
+    });
   };
 
-  handleGoogleLogin = () => {
-    this.props.screenProps.handleGoogleLogin();
+  handleGoogleLogin = async () => {
+    const userInfo = await handleGoogleLogin();
+    this.props.user.setUser({
+      ...userInfo,
+      first_name: userInfo.given_name,
```
Comment (thibmaek) [inline screens/login/LoginContainer.js]: snake case key 😱

File: state/UserContainer.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    this.getFromStorage();
+  }
+
+  state = {};
+
+  async getFromStorage() {
+    const result = await AsyncStorage.getItem(USER_STORAGE);
+
+    if (result) {
+      // console.log('Read AsyncStorage', USER_STORAGE, result);
```
Comment (mikeverf) [inline state/UserContainer.js]: Comment :)

File: App.js (hunk 3 above)
Comment (easybird) [reply App.js]: trash from during dev, needed  cleanup idd!

File: App.js (hunk 1 above)
Comment (easybird) [reply App.js]: no local userInfo needed anymore

File: screens/personalSchedule/PersonalScheduleScreen.js [hunk 7]
```diff
@@ -42,10 +42,10 @@ const styles = StyleSheet.create({
   },
 });
 
-const PersonalScheduleScreen = ({ sessions, handleGoBack }) => (
+const PersonalScheduleScreen = ({ sessions, handleGoBack, firstName }) => (
   <View style={styles.container}>
     <View style={styles.header}>
-      <Text style={styles.title}>My Schedule</Text>
+      <Text style={styles.title}>{`Your Schedule, ${firstName}`}</Text>
```
Comment (thibmaek) [inline screens/personalSchedule/PersonalScheduleScreen.js]: Cool!

File: screens/screenConfig.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
+import { StackNavigator } from 'react-navigation';
+import { HomeContainer, LoginContainer, PersonalScheduleContainer } from './';
+
+export const HOME = 'Home';
+export const PERSONAL_SCHEDULE = 'PersonalSchedule';
+export const LOGIN = 'Login';
+export const MAIN = 'Main';
+
+const createMainNavigator = initialRouteName =>
```
Comment (thibmaek) [inline screens/screenConfig.js]: Needs argument type & return type

File: state/UserContainer.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+type UserState = ?TUser;
+
+export default class UserContainer extends Container<UserState> {
+  constructor() {
+    super();
+    this.getFromStorage();
+  }
+
+  state = {};
```
Comment (thibmaek) [inline state/UserContainer.js]: Is this needed? Empty state doesn't need to be constructed I think

File: state/UserContainer.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+  async updateStorage() {
+    try {
+      await AsyncStorage.setItem(USER_STORAGE, JSON.stringify(this.state));
+    } catch (error) {
+      console.error('Error updating storage', error);
+    }
+  }
+
+  isAuthenticatedUser() {
```
Comment (thibmaek) [inline state/UserContainer.js]: arrow fn?

File: App.js (hunk 1 above)
Comment (thibmaek) [reply App.js]: Yeah but shiftData is back?

**Comment by devopsITP** [comment]
:shipit: This branch has been deployed to:
  https://expo.io/@itpocket/white-label-event-mobile-app-feature-28-add-autologin

  Download the [Expo](https://expo.io/) app and scan this QR code to get started!

  QR Code

File: App.js [hunk 11]
```diff
@@ -29,15 +29,15 @@ export default class App extends Component<*, State> {
...
+  componentWillMount() {
     initializeFirebase();
     const isLoggedIn = user.isAuthenticatedUser();
 
     this.RootStack = isLoggedIn ? createRootStackNavigator(HOME) : createRootStackNavigator(LOGIN);
   }
 
   componentWillUnmount() {
-    // TODO move to HomeContainer
+    // TODO #44 move to HomeContainer
```
Comment (thibmaek) [inline App.js]: 😍

**Comment by devopsITP** [comment]
:shipit: This branch has been deployed to:
  https://expo.io/@itpocket/white-label-event-mobile-app-feature-28-add-autologin

  Download the [Expo](https://expo.io/) app and scan this QR code to get started!

  QR Code

**Pull request merged**